webapp/src/selectors.ts

```diff
@@ -55,14 +51,14 @@ export const getSidebarData = createSelector(
         const {username, sidebarContent, reviewDetails, yourPrDetails, organization, rhsState} = pluginState;
         return {
             username,
-            reviews: mapPrsToDetails(sidebarContent.reviews || emptyArray, reviewDetails),
-            yourPrs: mapPrsToDetails(sidebarContent.prs || emptyArray, yourPrDetails),
-            yourAssignments: sidebarContent.assignments || emptyArray,
-            unreads: sidebarContent.unreads || emptyArray,
+            reviews: mapPrsToDetails(sidebarContent.reviews || [], reviewDetails),
+            yourPrs: mapPrsToDetails(sidebarContent.prs || [], yourPrDetails),
+            yourAssignments: sidebarContent.assignments || [],
+            unreads: sidebarContent.unreads || [],
```

Comment:
Can we have the return values of the two exported functions be typed?

Reply:
@Kshitij-Katiyar Right, I suppose it is just one function. I meant to have a separate type for the `{username, reviews...}` object I think

Also for the empty arrays here, can we create a single reference to an empty array to potentially avoid unnecessary referential changes in the component?

```ts
const emptyArray = [];

function mapPrsToDetails(prs: GithubIssueData[], details: PrsDetailsData[]) {

reviews: mapPrsToDetails(sidebarContent.reviews || emptyArray, reviewDetails),
```

Also wondering why we need to do the `|| []` in the first place. Can we instead have the backend make sure the arrays are all there? Not a requirement though